### ISSUE TITLE
CSIS Acceptance Test by METEOGRID

### ISSUE THREAD

**Issue by p-a-s-c-a-l** [clarity-h2020/csis-technical-validation#54]
As announced [here](https://github.com/clarity-h2020/csis-technical-validation/issues/1), we now start with the CSIS Acceptance Tests.

Please read the [Acceptance Test Specification](https://github.com/clarity-h2020/csis-technical-validation/wiki/Acceptance-Test-Specification) and follow the 11 steps in the **Walkthrough** section to test the [public BETA of CSIS](https://csis.myclimateservice.eu/).

Please report any problems or bugs according to the instructions in the **Giving Feedback** section.

If you have further questions regarding the general test process, please contact @therter and @p-a-s-c-a-l

**Comment by therter** [comment]
Have you encountered any problems/bugs?

**Comment by ghilbrae** [comment]
URL -> https://csis.myclimateservice.eu/group/169/edit/trigger?destination=/study/169/view/summary
Date & Time -> 2020-06-22 10:30:00
Browser Version and Operating System -> Chrome Version 83.0.4103.61 (Official Build) (64-bit) / OS - Linux 4.14.0-3-amd64 #1 SMP Debian 4.14.17-1 (2018-02-14) x86_64 GNU/Linux

An error triggered upon clicking on the button in the dialog to Start Calculations:

```
The website encountered an unexpected error. Please try again later.

Drupal\Core\Entity\EntityStorageException: Exception in Rules: 
study templates[rules_study_templates]: SQLSTATE[22P02]: 
Invalid text representation: 7 ERROR: invalid input syntax for integer: "{{study_group_study_type}}" 
LINE 6: WHERE (((taxonomy_term_field_data.tid = '{{study_group_study... ^: SELECT 
node_field_data_taxonomy_term__field_gl_template.langcode AS 
node_field_data_taxonomy_term__field_gl_template_langcode, taxonomy_term_field_data.tid AS tid, 
node_field_data_taxonomy_term__field_gl_template.nid AS 
node_field_data_taxonomy_term__field_gl_template_nid FROM {taxonomy_term_field_data} 
taxonomy_term_field_data LEFT JOIN {taxonomy_term__field_gl_template} 
taxonomy_term__field_gl_template ON taxonomy_term_field_data.tid = 
taxonomy_term__field_gl_template.entity_id AND taxonomy_term__field_gl_template.deleted = 
:views_join_condition_0 INNER JOIN {node_field_data} 
node_field_data_taxonomy_term__field_gl_template ON 
taxonomy_term__field_gl_template.field_gl_template_target_id = 
node_field_data_taxonomy_term__field_gl_template.nid WHERE (((taxonomy_term_field_data.tid = 
:taxonomy_term_field_data_tid)) AND ((taxonomy_term_field_data.status = 
:db_condition_placeholder_1) AND (taxonomy_term_field_data.vid IN 
(:db_condition_placeholder_2)))) AND (EXISTS (SELECT na.nid AS nid FROM {node_access} na 
WHERE (((gid IN (:db_condition_placeholder_3)) AND (realm = :db_condition_placeholder_4)) OR 
((gid IN (:db_condition_placeholder_5, :db_condition_placeholder_6, :db_condition_placeholder_7, 
:db_condition_placeholder_8, :db_condition_placeholder_9, :db_condition_placeholder_10, 
:db_condition_placeholder_11, :db_condition_placeholder_12, :db_condition_placeholder_13, 
:db_condition_placeholder_14, :db_condition_placeholder_15, :db_condition_placeholder_16, 
:db_condition_placeholder_17, :db_condition_placeholder_18, :db_condition_placeholder_19, 
:db_condition_placeholder_20, :db_condition_placeholder_21, :db_condition_placeholder_22, 
[...]
:db_condition_placeholder_436, :db_condition_placeholder_437, :db_condition_placeholder_438, 
:db_condition_placeholder_439, :db_condition_placeholder_440, :db_condition_placeholder_441, 
:db_condition_placeholder_442, :db_condition_placeholder_443, :db_condition_placeholder_444)) 
AND (realm = :db_condition_placeholder_445))) AND (na.grant_view >= 
:db_condition_placeholder_446) AND (na.fallback = :db_condition_placeholder_447) AND 
((node_field_data_taxonomy_term__field_gl_template.nid = na.nid)))) LIMIT 11 OFFSET 0; Array ( 
[:taxonomy_term_field_data_tid] => {{study_group_study_type}} [:db_condition_placeholder_1] => 1 
[:db_condition_placeholder_2] => study_type [:db_condition_placeholder_3] => 0 
[:db_condition_placeholder_4] => all [:db_condition_placeholder_5] => 142 
[:db_condition_placeholder_6] => 143 [:db_condition_placeholder_7] => 145 
[...]
[:db_condition_placeholder_435] => 164 [:db_condition_placeholder_436] => 149 
[:db_condition_placeholder_437] => 151 [:db_condition_placeholder_438] => 153 
[:db_condition_placeholder_439] => 150 [:db_condition_placeholder_440] => 152 
[:db_condition_placeholder_441] => 88 [:db_condition_placeholder_442] => 154 
[:db_condition_placeholder_443] => 95 [:db_condition_placeholder_444] => 132 
[:db_condition_placeholder_445] => gnode_unpublished:report_image 
[:db_condition_placeholder_446] => 1 [:db_condition_placeholder_447] => 1 
[:views_join_condition_0] => 0 ) in Drupal\Core\Entity\Sql\SqlContentEntityStorage->save() (line 846 
of core/lib/Drupal/Core/Entity/Sql/SqlContentEntityStorage.php).
```

Complete error: 
[error_start_calculations.txt](https://github.com/clarity-h2020/csis-technical-validation/files/4812211/error_start_calculations.txt)

It seems to work the second time that I tried to launch the process.

**Comment by ghilbrae** [comment]
URL -> https://csis.myclimateservice.eu/study/169/step/4644/view/table
Date & Time -> 2020-06-23 8:39:00
Browser Version and Operating System -> Chrome Version 83.0.4103.61 (Official Build) (64-bit) / OS - Linux 4.14.0-3-amd64 #1 SMP Debian 4.14.17-1 (2018-02-14) x86_64 GNU/Linux

The table component does not load. The error on the JS console is the following: `no field field_basic_auth_credentials in user profile angela`

**Comment by ghilbrae** [comment]
URL -> https://csis.myclimateservice.eu/study/169/step/4644/view/table
Date & Time -> 2020-06-23 8:42:00
Browser Version and Operating System -> Chrome Version 83.0.4103.61 (Official Build) (64-bit) / OS - Linux 4.14.0-3-amd64 #1 SMP Debian 4.14.17-1 (2018-02-14) x86_64 GNU/Linux

On occasion the layers won't load completely:

![image](https://user-images.githubusercontent.com/1615745/85369505-82277080-b52d-11ea-9e45-48821503dde4.png)

Tested on both maps on several tabs.

**Comment by ghilbrae** [comment]
URL -> https://csis.myclimateservice.eu/study/169/step/4648/view/scenario-analysis
Date & Time -> 2020-06-23 8:50:00
Browser Version and Operating System -> Chrome Version 83.0.4103.61 (Official Build) (64-bit) / OS - Linux 4.14.0-3-amd64 #1 SMP Debian 4.14.17-1 (2018-02-14) x86_64 GNU/Linux

The Scenario Analysis tab presents an error and no action can be performed:

```
{"data":"<html><head><title>Apache Tomcat/6.0.20 - Error report</title><style><!--H1 
{font-family:Tahoma,Arial,sans-serif;color:white;background-color:#525D76;font-size:22px;} 
H2 {font-family:Tahoma,Arial,sans-serif;color:white;background-color:#525D76;font-
size:16px;} H3 {font-family:Tahoma,Arial,sans-serif;color:white;background-
color:#525D76;font-size:14px;} BODY {font-family:Tahoma,Arial,sans-
serif;color:black;background-color:white;} B {font-family:Tahoma,Arial,sans-
serif;color:white;background-color:#525D76;} P {font-family:Tahoma,Arial,sans-
serif;background:white;color:black;font-size:12px;}A {color : black;}A.name {color : 
black;}HR {color : #525D76;}--></style> </head><body><h1>HTTP Status 401 - 
Unauthorized</h1><HR size=\"1\" noshade=\"noshade\"><p><b>type</b> Status 
report</p><p><b>message</b> <u>Unauthorized</u></p><p><b>description</b> 
<u>This request requires HTTP authentication (Unauthorized).</u></p><HR size=\"1\" 
noshade=\"noshade\"><h3>Apache Tomcat/6.0.20</h3></body>
</html>","status":401,"config":{"method":"GET","transformRequest":
[null],"transformResponse":[null],"headers":{"Authorization":"Basic 
dW5kZWZpbmVk","Accept":"application/json, text/plain, */*"},"url":"https://service.emikat.at
/EmiKatTst/api/scenarios/3493/feature/view.2994/table
/data","withCredentials":false},"statusText":"Unauthorized"}
```

![image](https://user-images.githubusercontent.com/1615745/85370262-b6e7f780-b52e-11ea-9020-dbd8b98b5413.png)

**Comment by p-a-s-c-a-l** [comment]
Thanks @ghilbrae !

> Complete error:
> [error_start_calculations.txt](https://github.com/clarity-h2020/csis-technical-validation/files/4812211/error_start_calculations.txt)

That's the same error as reported [here](https://github.com/clarity-h2020/csis-technical-validation/issues/51#issuecomment-647948828). So maybe @patrickkaleta could have a look at this.

**Comment by p-a-s-c-a-l** [comment]
> The table component does not load. The error on the JS console is the following: `no field field_basic_auth_credentials in user profile angela`

You are probably using an "old" account. Did you login via CAS  (https://profile.myclimateservices.eu/cas/login?service=https%3A//csis.myclimateservice.eu/casservice) or directly onCIS (https://csis.myclimateservice.eu/user/login)? What's your user name?

**Comment by ghilbrae** [comment]
> You are probably using an "old" account. Did you login via CAS (https://profile.myclimateservices.eu/cas/login?service=https%3A//csis.myclimateservice.eu/casservice) or directly on CIS (https://csis.myclimateservice.eu/user/login)? What's your user name?

I'm using the account I've been using since the beginning, so before CAS was implemented. My username is *angela*.

**Comment by patrickkaleta** [comment]
> > You are probably using an "old" account. Did you login via CAS (https://profile.myclimateservices.eu/cas/login?service=https%3A//csis.myclimateservice.eu/casservice) or directly on CIS (https://csis.myclimateservice.eu/user/login)? What's your user name?
> 
> I'm using the account I've been using since the beginning, so before CAS was implemented. My username is _angela_.

That's an old account, for which we forgot to manually add the `basic_auth_credential` field. I've fixed that. @ghilbrae it should work now.

**Comment by ghilbrae** [comment]
Thanks! @patrickkaleta

**Comment by p-a-s-c-a-l** [comment]
> I've fixed that. @ghilbrae it should work now.

Thanks! So now @ghilbrae can continue with the tests :-)

**Comment by ghilbrae** [comment]
URL -> https://csis.myclimateservice.eu/study/169/step/4648/view/summary
Date & Time -> 2020-06-25 10:56:00
Browser Version and Operating System -> Chrome Version 83.0.4103.61 (Official Build) (64-bit) / OS - Linux 4.14.0-3-amd64 #1 SMP Debian 4.14.17-1 (2018-02-14) x86_64 GNU/Linux

The table included in the report appears under Attached Maps instead of under Attached Tables. The image show this happening in Risk and Impact Assessment but it happens everywhere where this is an option.

![image](https://user-images.githubusercontent.com/1615745/85689082-a07aa100-b6d2-11ea-9019-4701501171d6.png)

**Comment by ghilbrae** [comment]
URL -> https://csis.myclimateservice.eu/study/169/step/4649/view/table
Date & Time -> 2020-06-25 11:05:00
Browser Version and Operating System -> Chrome Version 83.0.4103.61 (Official Build) (64-bit) / OS - Linux 4.14.0-3-amd64 #1 SMP Debian 4.14.17-1 (2018-02-14) x86_64 GNU/Linux
The check boxes are not aligned with the options while selecting entities for Adaptation Projects

![image](https://user-images.githubusercontent.com/1615745/85690293-bb014a00-b6d3-11ea-89e7-7d1f7294545d.png)

**Comment by ghilbrae** [comment]
Analysis Twins (https://csis.myclimateservice.eu/study/169/step/4649/view/twins) do not seem to work. Tried several distances with no luck and also selected adaptation options from the table as specified on the Twins description and nothing seems to change.

![image](https://user-images.githubusercontent.com/1615745/85692300-93ab7c80-b6d5-11ea-9d43-a0bb70a500ef.png)

**Comment by patrickkaleta** [comment]
> Analysis Twins (https://csis.myclimateservice.eu/study/169/step/4649/view/twins) do not seem to work. Tried several distances with no luck and also selected adaptation options from the table as specified on the Twins description and nothing seems to change.
> 
> ![image](https://user-images.githubusercontent.com/1615745/85692300-93ab7c80-b6d5-11ea-9d43-a0bb70a500ef.png)

I don't know how relevant the current **Twins** Tab is for the **Adaptation Option Identification**. We had to greatly change the conpect of Adaptation options and are now using so-called Adaptation Strategies inside Studies. I'm sure we will therefore have to adapt the Twins concept for this GL-step as well. So for now it's probably best to hide the **Twins** tab in this step.

Also, I noticed that there is an **External** Tab in your Study for this GL-Step. I don't think that should be there (what would we show in there?), so I think we have to remove this from the GL-templates of the Study Type "Advanced Screening: Urban Infrastructure"
